File: check_template.py
import re

with open('/Users/mac/Desktop/saas-admin-workbench/src/views/Setting/Team.vue', 'r') as f:
    content = f.read()

start = content.find('<template>')
end = content.find('</template>', start) + len('</template>')
template = content[start:end]

stack = []
line_num = 1
errors = []

i = 0
while i < len(template):
    if template[i] == '\n':
        line_num += 1
    if template[i] == '<' and template[i:i+4] != '<!--':
        tag_end = template.find('>', i)
        if tag_end == -1:
            break
        tag = template[i+1:tag_end].strip()

        if tag.startswith('!'):
            i = tag_end + 1
            continue

        is_closing = tag.startswith('/')
        is_self_closing = tag.endswith('/')

        if is_closing:
            parts = tag[1:].strip().split()
            tag_name = parts[0] if parts else ''
            if stack and stack[-1] == tag_name:
                stack.pop()
            else:
                errors.append(f'Line {line_num}: mismatched </{tag_name}>')
        elif is_self_closing:
            pass
        else:
            parts = tag.split()
            tag_name = parts[0] if parts else ''
            if tag_name not in ['template', 'slot', 'component', 'el-dropdown-menu', 'el-tooltip']:
                stack.append(tag_name)

        i = tag_end + 1
    else:
        i += 1

if stack:
    for t in stack:
        errors.append(f'Unclosed tag: {t}')

print(f'Total errors: {len(errors)}')
for e in errors[:10]:
    print(e)
if not errors:
    print('Template structure is VALID!')

File: check_template.mjs
import { readFileSync } from 'fs';

const content = readFileSync('src/views/Setting/Team.vue', 'utf8');
const start = content.indexOf('<template>');
const end = content.indexOf('</template>', start) + '</template>'.length;
const template = content.slice(start, end);

const selfClosing = new Set([
  'img', 'br', 'hr', 'input', 'meta', 'link', 'area', 'base', 'col', 'embed', 'param', 'source', 'track', 'wbr',
  'path', 'circle', 'rect', 'line', 'polyline', 'polygon', 'ellipse', 'symbol', 'defs', 'use', 'stop', 'feBlend', 'feColorMatrix', 'feComponentTransfer', 'feComposite', 'feConvolveMatrix', 'feDiffuseLighting', 'feDisplacementMap', 'feDistantLight', 'feFlood', 'feFuncA', 'feFuncB', 'feFuncG', 'feFuncR', 'feGaussianBlur', 'feImage', 'feMerge', 'feMergeNode', 'feMorphology', 'feOffset', 'fePointLight', 'feSpecularLighting', 'feSpotLight', 'feTile', 'feTurbulence', 'linearGradient', 'radialGradient', 'mask', 'clipPath', 'g', 'svg', 'line', 'text', 'tspan'
]);

const stack = [];
let errors = [];
let line = 1;

for (let i = 0; i < template.length; i++) {
  if (template[i] === '\n') line++;
  if (template[i] !== '<') continue;
  if (template[i+1] === '!') continue;

  let tagEnd = template.indexOf('>', i);
  if (tagEnd === -1) continue;

  let inner = template.slice(i+1, tagEnd).trim();

  if (inner === '--') { continue; }

  let isClosing = inner.startsWith('/');
  let tagName = (isClosing ? inner.slice(1) : inner).split(/\s/)[0].toLowerCase();

  if (isClosing) {
    if (stack.length === 0) {
      errors.push(`line ${line}: unexpected </${tagName}>`);
    } else if (stack[stack.length-1] !== tagName) {
      errors.push(`line ${line}: expected </${stack[stack.length-1]}> but got </${tagName}>`);
    } else {
      stack.pop();
    }
  } else if (inner.endsWith('/')) {
    // self-closing - don't push
  } else {
    stack.push(tagName);
  }

  i = tagEnd;
}

if (stack.length > 0) errors.push(`unclosed tags: ${stack.join(', ')}`);
console.log(`Errors: ${errors.length}`);
errors.slice(0, 10).forEach(e => console.log(e));
if (errors.length === 0) console.log('Template structure is VALID!');
